### ISSUE TITLE
[Bug] GetEnvironmentProperty returns the wrong SunDir that shifts the sun when setting it back in

### ISSUE THREAD

**Issue by Please-Pick-a-Name** [Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker#181]
### Describe the bug

when SunDir is predefined in map with a rotated Skybox, getting the SunDir via GetEnvironmentProperty returns the wrong value, and setting it back in will shift the sun location

this happened at when i make a copy of the https://steamcommunity.com/sharedfiles/filedetails/?id=2782455482, which has a predefined SunDir and a rotated skybox, and When i access the enviroments properties with my https://steamcommunity.com/sharedfiles/filedetails/?id=2633921854, the sun immediately changed location
i know my mod functions by getting the environment properties every tick and setting the changed properties back in, at the same tick, and if no changes happenes then it will set the properties that experienced no changes back in

### Steps to reproduce the behavior

1. Go to map editor
2. make a map with rotated skybox and predefined SunDir
3. prepare a script that get the SunDir properties and set whatever it gets back in, maybe on a keypress
4. load the map, press the key
5. see sun suddenly moves

### Expected behavior

gets correct SunDir

### Environment

- Teardown Version: 0.9.5
- OS: Windows
- GPU:
- Graphics driver version:
- CPU:

**Comment by nooitaf** [comment]
To give some more context .. 

This is what it's set to:
![image](https://user-images.githubusercontent.com/1898194/160219748-239169e1-d748-46c8-9a93-05c1b18b5de2.png)
```
x  -0.05
y   0.2
z  -1.0
```

This is what you get with ..
```lua
  x, y, z = GetEnvironmentProperty("sunDir")
  print(x)
  print(y)
  print(z)
```
```shell
x -0.97940415143967
y  0.19588083028793
z  0.048970270901918
```
They seem to be in reversed order, and `x` -> `z` is negative flipped?

**Comment by Please-Pick-a-Name** [comment]
> To give some more context ..
> 
> This is what it's set to: ![image](https://user-images.githubusercontent.com/1898194/160219748-239169e1-d748-46c8-9a93-05c1b18b5de2.png)
> 
> ```
> x  -0.05
> y   0.2
> z  -1.0
> ```
> 
> This is what you get with ..
> 
> ```lua
>   x, y, z = GetEnvironmentProperty("sunDir")
>   print(x)
>   print(y)
>   print(z)
> ```
> 
> ```shell
> x -0.97940415143967
> y  0.19588083028793
> z  0.048970270901918
> ```
> 
> They seem to be in reversed order, and `x` -> `z` is negative flipped?

oh they are just fliped? i thought it was cause by the skyboxrot lol, should have done more testing

**Comment by tuxedolabs** [comment]
There seems to be an issue wehn combining with skybox rotation. There is no easy fix, since a fix would likely affect existing content. I'll think about it. Good find!

**Comment by Please-Pick-a-Name** [comment]
> There seems to be an issue wehn combining with skybox rotation. There is no easy fix, since a fix would likely affect existing content. I'll think about it. Good find!

yeah, it seems that rotated skybox is already rare enough, let along skybox that needs predefined SunDir, and then someone needs to try edit that enviroment skybox at run time, then they would have a chance of finding this out, lol, as they might just think its a value out of bound for the api but not the editor causing some funniness

**Comment by gaberundlett-tuxedo** [comment]
Maybe its unintuitive, but through testing, it appears that the function `GetEnvironmentProperty("sunDir")` gives the actual sun direction vector, as opposed to the value that's written in the editor. This is apparent when you realize that it's normalized as well. Closing as resolved.
